File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/validation/validator/UniqueUserValidator.java
package com.windea.demo.cloudcollect.core.validation.validator;

import com.windea.demo.cloudcollect.core.domain.entity.User;
import com.windea.demo.cloudcollect.core.service.UserService;
import com.windea.demo.cloudcollect.core.validation.annotation.UniqueUser;
import org.springframework.beans.factory.annotation.Autowired;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * 唯一用户的校验器。
 */
public class UniqueUserValidator implements ConstraintValidator<UniqueUser, User> {
	@Autowired private UserService service;

	@Override
	public boolean isValid(User value, ConstraintValidatorContext context) {
		return !service.exists(value);
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/validation/validator/UniqueUserValidator.kt
package com.windea.demo.cloudcollect.core.validation.validator

import com.windea.demo.cloudcollect.core.domain.entity.*
import com.windea.demo.cloudcollect.core.service.*
import com.windea.demo.cloudcollect.core.validation.annotation.*
import org.springframework.beans.factory.annotation.*
import javax.validation.*

/**唯一用户的校验器。*/
open class UniqueUserValidator : ConstraintValidator<UniqueUser, User> {
	@Autowired private lateinit var userService: UserService
	
	override fun isValid(value: User, context: ConstraintValidatorContext): Boolean {
		return !userService.existsByUsernameOrEmail(value.username, value.email)
	}
}
